workers/data_refinery_workers/processors/array_express.py

```diff
@@ -11,97 +12,114 @@
 logger = get_task_logger(__name__)
 
 
-PACKAGE_NAME_CORRECTIONS = {
-    "hugene10stv1hsentrezgprobe": "hugene10sthsentrezgprobe"
-}
+def _prepare_files(job_context: Dict) -> Dict:
+    """Moves the .CEL file from the raw directory to the temp directory
 
-
-def cel_to_pcl(kwargs: Dict) -> Dict:
-    """Process .CEL files to .PCL format using R.
-
-    Moves the .CEL file from the raw directory to the temp directory,
-    calls ProcessCelFiles, uploads the processed file, and cleans up
-    the raw/temp files. Because it uses the file_management module
-    this works seamlessly whether S3 is being used or not.
+    Also adds the keys input_file and output_file to job_context so
+    everything is prepared for processing.
     """
-    # Array Express processor jobs have one batch per job.
-    batch = kwargs["batches"][0]
+    # Array Express processor jobs have only one batch per job.
+    batch = job_context["batches"][0]
 
     try:
         file_management.download_raw_file(batch)
     except Exception:
-        logging.exception("Exception caught while retrieving %s for batch %d during Job #%d.",
+        logging.exception(("Exception caught while retrieving raw file "
+                           "%s for batch %d during Processor Job #%d."),
                           file_management.get_raw_path(batch),
-                          batch.id,
-                          kwargs["job_id"])
-        kwargs["success"] = False
-        return kwargs
+                          batch.id, job_context["job_id"])
+        failure_template = "Exception caught while retrieving raw file {}"
+        job_context["job"].failure_reason = failure_template.format(batch.name)
+        job_context["success"] = False
+        return job_context
+
+    job_context["input_file"] = file_management.get_temp_pre_path(batch)
+    job_context["output_file"] = file_management.get_temp_post_path(batch)
+    return job_context
 
-    input_file = file_management.get_temp_pre_path(batch)
-    output_file = file_management.get_temp_post_path(batch)
 
-    header = ro.r['::']('affyio', 'read.celfile.header')(input_file)
+def _determine_brainarray_package(job_context: Dict) -> Dict:
+    """Determines the right brainarray package to use for the file.
+
+    Expects job_context to contain the key 'input_file'. Adds the key
+    'brainarray_package' to job_context."""
+    input_file = job_context["input_file"]
+    try:
+        header = ro.r['::']('affyio', 'read.celfile.header')(input_file)
+    except RRuntimeError as e:
+        # Array Express processor jobs have only one batch per job.
+        file_management.remove_temp_directory(job_context["batches"][0])
+
+        base_error_template = "unable to read Affy header in input file {0} due to error: {1}"
+        base_error_message = base_error_template.format(input_file, str(e))
+        log_message = "Processor Job %d running AFFY_TO_PCL pipeline " + base_error_message
+        logger.error(log_message)
```

Comment:
Yep, yep I do. (I wasn't as careful about testing these log statements as I should have been because the next branch I worked on was about fixing log statements.)